src/validation.cpp

```diff
@@ -3925,7 +3925,7 @@ static bool ContextualCheckBlock(const CBlock& block, CValidationState& state, c
     {
         CScript expect = CScript() << nHeight;
         if (block.vtx[0]->vin[0].scriptSig.size() < expect.size() ||
-            !std::equal(expect.begin(), expect.end(), block.vtx[0]->vin[0].scriptSig.begin())) {
+            !ranges::equal(expect, block.vtx[0]->vin[0].scriptSig)) {
```

Comment:
This should be reverted because `block.vtx[0]->vin[0].scriptSig.size()` can be `>= expect.size()`, not just `==`
```suggestion
            !std::equal(expect.begin(), expect.end(), block.vtx[0]->vin[0].scriptSig.begin())) {
```